x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -359,6 +341,16 @@ private void notifyModelMemoryStatusChange(ModelSizeStats modelSizeStats) {
         }
     }
 
+    private void notifyCategorizationStatusChange(ModelSizeStats modelSizeStats) {
+        ModelSizeStats.CategorizationStatus categorizationStatus = modelSizeStats.getCategorizationStatus();
+        if (categorizationStatus != latestModelSizeStats.getCategorizationStatus()) {
```

Comment:
Should we also emit auditor warning if the `categorizationStatus` is WARN in the first `modelSizeStats` analyzed?

Reply:
> Should we also emit auditor warning if the `categorizationStatus` is WARN in the first `modelSizeStats` analyzed?

Yes, that should work because a default constructed `ModelSizeStats` will have `CategorizationStatus.OK`.  But I will add a test to prove it.